orchestrator-salt/src/main/resources/salt/salt/metering/scripts/metering_package_manager.sh

```diff
@@ -0,0 +1,316 @@
+      download_binary "$METERING_SERVICE_DELIVERY_URL" "$METERING_FILE_NAME" "true"
+    else
+      do_exit 1 "Cannot download Metering RPM file (from $archive_url or $METERING_SERVICE_DELIVERY_URL)."
+    fi
+  fi
+  if [[ "$OPERATION_NAME" == "upgrade-only" ]]; then
+    if [[ -d "/srv/salt/metering" ]]; then
+      mkdir -p /srv/salt/metering/files
+      cp "$WORKING_DIR/$METERING_FILE_NAME" /srv/salt/metering/files/$METERING_FILE_NAME
+    fi
+  fi
+}
+
+function download_binary() {
+    local rpm_url=${1:?"usage: <rpm_url>"}
+    local rpm_name=${2:?"usage: <rpm_name>"}
+    local override=${3}
+    local download_file_path="$WORKING_DIR/${rpm_name}"
+    if [[ -f $download_file_path && "$override" == "" ]]; then
+      log "RPM file '$download_file_path' already exists. Download will be skipped."
+      return
+    fi
+    log "Downloading $rpm_url"
+    content_length=$(curl ${CURL_EXTRA_PARAMS} -s --head -k -L "${rpm_url}" | grep -i Content-Length | awk '{print $2}' | tr -d '\r' | tr -d '\n')
+    log "Content length: ${content_length}"
+    free_space="$(df -B1 --output=avail "$WORKING_DIR" | grep -v Avail)"
+    log "Free space: ${free_space}"
+    local fs=$(expr $free_space + 0)
+    local cl=$(expr $content_length + 0)
+    if [ "$fs" -gt "$cl" ]; then
+      log "Have enough space for download the rpm from: $rpm_url"
+    else
+      do_exit 1 "Not enough space to download the rpm from: $rpm_url)"
+    fi
+    log "Run command: curl -k -L --output $download_file_path $rpm_url"
+    curl ${CURL_EXTRA_PARAMS} -k -L --output "$download_file_path" "$rpm_url"
+    if [[ ! -f $download_file_path ]]; then
+      do_exit 1 "RPM file was not downloaded: $download_file_path. Exiting ..."
+    fi
+}
+
+function run_operation() {
+  while [[ $# -gt 0 ]]
+    do
+      key="$1"
+      case $key in
+        -d|--from-date)
+          FROM_DATE="$2"
+          shift 2
+        ;;
+        -f|--rpm-file)
+          LOCAL_RPM_FILE="$2"
+          shift 2
+        ;;
+        -u|--custom-rpm-url)
+          CUSTOM_RPM_URL="$2"
+          shift 2
+        ;;
+        -w|--working-dir)
+          WORKING_DIR="$2"
+          shift 2
+        ;;
+        *)
+          echo "Unknown option: $1"
+          do_exit 1
+        ;;
+      esac
+  done
+  init_logfile
+  init_salt_prefix
+  if [[ "$WORKING_DIR" == "" ]]; then
+    WORKING_DIR="/tmp"
+  elif [[ ! -d "$WORKING_DIR" ]]; then
+    info "Working directory does not exists. Creating it..."
+    mkdir -p "$WORKING_DIR"
+  fi
+  if [[ "$OPERATION_NAME" == "download" ]]; then
+    init_proxy_configs
+    if [[ "$https_proxy" != "" ]]; then
+      export https_proxy="${https_proxy}"
+      info "Found https_proxy settings."
+    fi
+    if [[ "$no_proxy" != "" ]]; then
+      export no_proxy="${no_proxy}"
+      info "Found no_proxy settings: $no_proxy"
+    fi
+    download
+    LOCAL_RPM_FILE="$WORKING_DIR/$METERING_FILE_NAME"
+    check_download_rpm_by_size
```

Comment:
why not just pass the file location as a parameter? it should be returned by the download method